core/unix/signal_linux.c

```diff
@@ -364,6 +364,8 @@ handle_pre_extended_syscall_sigmasks(dcontext_t *dcontext, kernel_sigset_t *sigm
     if (sizemask != sizeof(kernel_sigset_t))
         return false;
     ASSERT(!info->pre_syscall_app_sigprocmask_valid);
+    if (sigmask == NULL)
+        return true;
```

Comment:
I intentionally put the sizemask check there. I can see why the kernel might ignore the "wrong" size, but I don't see it specified anywhere, do you? 99.99% of the time I would expect glibc passing in the right size, moreover passing in the wrong size would be a bug in the app. Do we really need to write assembler code to test this?

Reply:
I think this is obsolete since we're not calling the handler any longer.

Reply:
You could argue that the man page implies that sizemask is ignored if sigmask is NULL but it is not 100% clear:

> If the sigmask argument is specified as NULL, then no signal mask manipulation is performed (and thus ppoll() differs from poll() only in the precision of the timeout argument).

We try to match kernel behavior as much as possible in case apps depend on it (it happens more often than we'd like).  I could see an app hand-rolling a syscall (in a JIT or sthg) and just setting mask and size to 0 and expecting it to work, because it does.  There are other syscalls where we've hit problems by not emulating the kernel on what seem like corner cases and instead going on intuition: sigaltstack, sigaction.  Sure, skip the test since it's non-trivial, but let's put the errors in the same order as the kernel.